tests/integ/modin/test_merge_asof.py

```diff
@@ -0,0 +1,388 @@
+                pd.Timestamp("2016-05-25 13:30:00.038"),
+                pd.Timestamp("2016-05-25 13:30:00.048"),
+                pd.Timestamp("2016-05-25 13:30:00.048"),
+                pd.Timestamp("2016-05-25 13:30:00.048"),
+            ],
+            "price": [51.95, 51.95, 720.77, 720.92, 98.0],
+            "quantity": [75, 155, 100, 100, 100],
+        }
+    )
+    return quotes, trades
+
+
+allow_exact_matches = pytest.mark.parametrize("allow_exact_matches", [True, False])
+direction = pytest.mark.parametrize("direction", ["backward", "forward"])
+
+
+@pytest.mark.parametrize("on", ["a", ["a"]])
+@allow_exact_matches
+@direction
+@sql_count_checker(query_count=1, join_count=1)
+def test_merge_asof_on(left_right_native_df, on, allow_exact_matches, direction):
+    left_native_df, right_native_df = left_right_native_df
+    left_snow_df, right_snow_df = pd.DataFrame(left_native_df), pd.DataFrame(
+        right_native_df
+    )
+    native_output = native_pd.merge_asof(
+        left_native_df,
+        right_native_df,
+        on=on,
+        allow_exact_matches=allow_exact_matches,
+        direction=direction,
+    )
+    snow_output = pd.merge_asof(
+        left_snow_df,
+        right_snow_df,
+        on=on,
+        allow_exact_matches=allow_exact_matches,
+        direction=direction,
+    )
+    assert_snowpark_pandas_equal_to_pandas(snow_output, native_output)
+
+
+@allow_exact_matches
+@direction
+@sql_count_checker(query_count=1, join_count=1)
+def test_merge_asof_on_switch_column_order(
+    left_right_native_df_switch_column_order, allow_exact_matches, direction
+):
+    left_native_df, right_native_df = left_right_native_df_switch_column_order
+    left_snow_df, right_snow_df = pd.DataFrame(left_native_df), pd.DataFrame(
+        right_native_df
+    )
+    native_output = native_pd.merge_asof(
+        left_native_df,
+        right_native_df,
+        on="a",
+        allow_exact_matches=allow_exact_matches,
+        direction=direction,
+    )
+    snow_output = pd.merge_asof(
+        left_snow_df,
+        right_snow_df,
+        on="a",
+        allow_exact_matches=allow_exact_matches,
+        direction=direction,
+    )
+    assert_snowpark_pandas_equal_to_pandas(snow_output, native_output)
+
+
+@allow_exact_matches
+@direction
+@sql_count_checker(query_count=1, join_count=1)
+def test_merge_asof_left_right_on(
+    left_right_native_df_left_right_on, allow_exact_matches, direction
+):
+    left_native_df, right_native_df = left_right_native_df_left_right_on
+    left_snow_df, right_snow_df = pd.DataFrame(left_native_df), pd.DataFrame(
+        right_native_df
+    )
+    native_output = native_pd.merge_asof(
+        left_native_df,
+        right_native_df,
+        left_on="a",
+        right_on="b",
+        allow_exact_matches=allow_exact_matches,
+        direction=direction,
+    )
+    snow_output = pd.merge_asof(
+        left_snow_df,
+        right_snow_df,
+        left_on="a",
```

Comment:
one test you want to add based on the implementation is probably when left_on = 'a', and right_on = 'a', which should have the same behavior as on = 'a'